File: structy/array-and-string/pair-sum.js
/*
Pair Sum
https://structy.net/problems/premium/pair-sum
27 August 2023
David Pratt
*/

const pairSum = (numbers, targetSum) => {
  // todo
  let result = [];
  for(let i = 0; i < numbers.length; i++){
    console.log(i);
    for(let j = i + 1; j <= numbers.length; j++){
      console.log(j);
      if(numbers[i] + numbers[j] === targetSum){
        result.push(i, j)
        break;
      }
    }
  }
  return result;
};

File: structy/array-and-string/pair-sum.py
'''
Pair Sum
https://structy.net/problems/premium/pair-sum
12 November 2023
David Pratt
'''


def pair_sum(numbers, target_sum):
  number_count = len(numbers)
  for i in range(number_count):
      for j in range(i + 1, number_count):
          if(numbers[i] + numbers[j] == target_sum):
              return (i, j);
  return False;
  
